src/names/name-resolution.md

```diff
@@ -1,4 +1,384 @@
+
+r[names.resolution.expansion.imports.ambiguity]
+#### Ambiguities
+
+r[names.resolution.expansion.imports.ambiguity.intro]
+Some situations are an error when there is an ambiguity as to which macro definition, `use` declaration, or module an import or macro invocation's name refers to. This happens when there are two name candidates that do not resolve to the same entity where neither candidate is [permitted] to shadow the other.
+
+r[names.resolution.expansion.imports.ambiguity.globvsglob]
+Names may not be resolved through ambiguous glob imports. Glob imports are allowed to import conflicting names in the same namespace as long as the name is not used. Names with conflicting candidates from ambiguous glob imports may still be shadowed by non glob imports and used without producing an error. The errors occur at time of use, not time of import.
+
+For example:
+
+```rust
+mod foo {
+    pub struct Qux;
+}
+
+mod bar {
+    pub struct Qux;
+}
+
+use foo::*;
+use bar::*; // OK, no name conflict.
+
+fn ambiguous_use() {
+    // This would be an error, due to the ambiguity.
+    //let x = Qux;
+}
+
+fn ambiguous_shadow() {
+    // This is permitted, since resolution is not through the ambiguous globs.
+    struct Qux;
+    let x = Qux;
+}
+```
+
+Multiple glob imports are allowed to import the same name, and that name is allowed to be used, if the imports are of the same item (following re-exports). The visibility of the name is the maximum visibility of the imports. For example:
+
+```rust
+mod foo {
+    pub struct Qux;
+}
+
+mod bar {
+    pub use super::foo::Qux;
+}
+
+// These both import the same `Qux`. The visibility of `Qux`
+// is `pub` because that is the maximum visibility between
+// these two `use` declarations.
+pub use bar::*;
+use foo::*;
+
+fn main() {
+    let _: Qux = Qux;
+}
+```
+
+r[names.resolution.expansion.imports.ambiguity.globvsouter]
+Names may not be resolved through glob imports when there is another candidate available in an [outer scope].
+
+```rust,compile_fail,E0659
+mod bar {
+    pub mod foo {
+//          ^-- glob `foo` candidate
+        pub struct Name;
+    }
+}
+
+pub mod foo {
+//      ^-- outer `foo` candidate
+    pub struct Name;
+}
+
+pub fn qux() {
+    use bar::*;
+    use foo::Name; // ERROR: `foo` is ambiguous
+}
+```
+
+```rust,compile_fail,E0659
+pub mod bar {
+    #[macro_export]
+    macro_rules! m {
+        () => {};
+    }
+
+    macro_rules! m2 {
+        () => {};
+    }
+    pub(crate) use m2 as m;
+}
+
+pub fn qux() {
+    use bar::*;
+    m!(); // ERROR: `m` is ambiguous
+}
+```
+
+> [!NOTE]
+> These ambiguity errors are specific to imports, even though they are only observed when those imports are used, having multiple candidates available for a given name during later stages of resolution is not considered an error, so long as none of the imports themselves are ambiguous, there will always be a single unambiguous closest resolution during later stages.
```

Comment:
Can this be split up into a few sentences?  (and the "during later stages" seems duplicated.)

Maybe something like:


```suggestion
> These ambiguity errors are specific to imports, even though they are only observed when those imports are used. Having multiple candidates available for a given name during later stages of resolution is not considered an error, so long as none of the imports themselves are ambiguous, there will always be a single unambiguous closest resolution.
```